app/src/main/java/com/projectfkklp/saristorepos/activities/user_profile/UserProfilePage.java

```diff
@@ -73,8 +100,49 @@ public void afterTextChanged(Editable editable) {
             }
         });
 
-        phoneText = findViewById(R.id.user_profile_phone);
-        gmailText = findViewById(R.id.user_profile_gmail);
+        phoneText.setText(currentUser.getPhoneNumber());
+        phoneText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                editUser.setPhoneUid(editable.toString());
+                updateButton.setEnabled(hasEdits());
+            }
+        });
+
+        gmailText.setText(currentUser.getGmail());
+        gmailText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable editable) {
+                editUser.setGmailUid(editable.toString());
+                updateButton.setEnabled(hasEdits());
+            }
+        });
+    }
```

Comment:
no need to for TextChangedListener, at all times we do not enable enable the text editor for this field